src/control/system/errors.go

```diff
@@ -40,6 +41,15 @@ func IsEmptyGroupMap(err error) bool {
 	return strings.Contains(errors.Cause(err).Error(), ErrEmptyGroupMap.Error())
 }
 
+// IsUninitialized returns a boolean indicating whether or not the
+// supplied error corresponds to an uninitialized system.
+func IsUninitialized(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(errors.Cause(err).Error(), ErrUninitialized.Error())
```

Comment:
his isn't ideal, it would be nice if we could match against the typed error

Reply:
The advice against matching on error strings applies to fixed string literals, e.g. `err.Error() == "some error message"`, as this is fragile and harder to maintain. In this case, we are providing a helper to match on an error even after it's been sent across gRPC and may have been mangled a bit.